backend/de.metas.manufacturing/src/main/sql/postgresql/system/30-de.metas.adempiere.libero/5681170_sys_gh14858_Order_By_SeqNo_PP_Product_BOM_Recursive_Excel_Report.sql

```diff
@@ -0,0 +1,138 @@
+/*
+ * #%L
+ * de.metas.manufacturing
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+drop function if exists PP_Product_BOM_Recursive(numeric, character varying);
+drop function if exists PP_Product_BOM_Recursive_Report(numeric);
+
+CREATE OR REPLACE FUNCTION pp_product_bom_recursive(p_pp_product_bom_id numeric, p_ad_language character varying)
+    RETURNS TABLE(line text, parent_line text, productvalue character varying, productname character varying, qtybom numeric, percentage numeric, uomsymbol character varying, depth integer, m_product_id numeric, isqtypercentage character, c_uom_id numeric, path integer[])
+    STABLE
+    LANGUAGE sql
+AS
+$$
+--
+	with recursive bomNode as (
+		(
+			select
+				array[1::integer] as path,
+				null::integer[] as parent_path,
+				1 as depth,
+				bomProduct.Value as ProductValue,
+				coalesce(pt.Name, bomProduct.Name) as ProductName,
+				bomProduct.M_Product_ID,
+				bomProduct.IsBOM,
+				bom.PP_Product_BOM_ID,
+				'N'::char(1) as IsQtyPercentage,
+				round(1::numeric, uom.StdPrecision) as QtyBOM,
+				null::numeric as Percentage,
+				COALESCE(uom.UOMSymbol, uomt.UOMSymbol) as UOMSymbol,
+				uom.C_UOM_ID
+			from PP_Product_BOM bom
+			inner join M_Product bomProduct on bomProduct.M_Product_ID=bom.M_Product_ID
+			LEFT OUTER JOIN M_Product_Trl pt    ON bomProduct.M_Product_ID = pt.M_Product_ID AND pt.AD_Language =p_ad_language
+       AND pt.isActive = 'Y'
+			left outer join C_UOM uom on uom.C_UOM_ID=coalesce(bom.C_UOM_ID, bomProduct.C_UOM_ID)
+			LEFT OUTER JOIN C_UOM_Trl uomt ON uom.C_UOM_ID = uomt.C_UOM_ID AND uomt.IsActive='Y' and uomt.AD_Language = p_ad_language
+			where
+			bom.PP_Product_BOM_ID=PP_Product_BOM_Recursive.p_PP_Product_BOM_ID
+		)
+		--
+		union all
+		--
+		(
+			select
+				parent.path || (row_number() over (partition by bomLine.PP_Product_BOM_ID order by bomLine.line, bomLine.PP_Product_BOMLine_ID))::integer as path,
```

Comment:
before, the records were ordered only by PP_Product_BOMLine_ID, but I made it so that they will be ordered by line, and after that, by ID